src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -91,9 +97,27 @@ public bool ExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionConte
                 showProgress: true,
                 c =>
                 {
-                    var docOptions = document.GetOptionsAsync(c.CancellationToken).WaitAndGetResult(c.CancellationToken);
+                    var workspaceOptions = document.Project.Solution.Workspace.Options;
+
+                    // AB Test
+                    // If the feature is OFF and the feature options have not yet been set to their assigned flight, do so now
+                    if (!workspaceOptions.GetOption(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting, LanguageNames.CSharp)
+                        && !workspaceOptions.GetOption(CodeCleanupABTestOptions.ParticipatedInExperiment))
+                    {
+                        var experimentationService = document.Project.Solution.Workspace.Services.GetService<IExperimentationService>();
+
+                        if (experimentationService != null)
+                        {
+                            if (experimentationService.IsExperimentEnabled(s_experimentName))
+                            {
+                                workspaceOptions = workspaceOptions.WithChangedOption(new OptionKey(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting, LanguageNames.CSharp), true);
```

Comment:
use this: document.Project.Language instead of LanguageNames.CSharp

Reply:
Actually, I think we need to be setting both languages here. Otherwise, we'll turn the experiment on for one language but not the other.

---
In reply to: [203908788](https://github.com/dotnet/roslyn/pull/28625#discussion_r203908788) [](ancestors = 203908788)

Reply:
we only have code cleanup service for C# right now